File: blood_pressure/lib/models/doctor.dart
import 'dart:convert';

class doctor {
  String? doctorName;
  String? email;
  String? password;
  String? address;
  String? phonenumber;
  String? gender;
  String? doctorId;

  doctor({
    required this.doctorName,
    required this.email,
    required this.password,
    required this.address,
    required this.phonenumber,
    required this.gender,
    required this.doctorId,
  });
  Map<String, dynamic> toMap() {
    return {
      'doctorName': doctorName,
      'email': email,
      'password': password,
      'address': address,
      'phonenumber': phonenumber,
      'gender': gender,
      'doctorId': doctorId,
    };
  }

  doctor.fromJson(Map<String, dynamic> json) {
    if (json["doctorName"] is String) {
      doctorName = json["doctorName"];
    }
    if (json["email"] is String) {
      email = json["email"];
    }
    if (json["password"] is String) {
      password = json["password"];
    }
    if (json["address"] is String) {
      address = json["address"];
    }
    if (json["phonenumber"] is String) {
      phonenumber = json["phonenumber"];
    }
    if (json["gender"] is String) {
      gender = json["gender"];
    }
    if (json["_id"] is String) {
      doctorId = json["_id"];
    }
  }

  String toJson() => json.encode(toMap());
}

File: blood_pressure/server/models/Doctor.js
const mongoose = require('mongoose')

const Schema = mongoose.Schema

const Doctor = new Schema({
    doctorName: {type: String, required: true},
    email: {
        required: true,
        type: String,
        trim: true,
        validate: {
          validator: (value) => {
            const re =
              /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
            return value.match(re);
          },
          message: "Please enter a valid email address",
        },
      },
    password:{type:String, require:true},
    address:{type: String, },
    phonenumber: {type: String,},
    gender:{type:String,},
    schedule:[{type: mongoose.Schema.Types.ObjectId, ref:"schedule"}]
},
)

module.exports = mongoose.model('doctor', Doctor)